x-pack/plugins/apm/public/components/app/transaction_details/WaterfallWithSummmary/WaterfallContainer/WaterfallLegends.tsx

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { EuiTitle } from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+import React from 'react';
+import styled from 'styled-components';
+import { px, unit } from '../../../../../style/variables';
+import { Legend } from '../../../../shared/charts/Legend';
+import {
+  IWaterfallLegend,
+  WaterfallLegendType,
+} from './Waterfall/waterfall_helpers/waterfall_helpers';
+
+const Legends = styled.div`
+  display: flex;
+
+  > * {
+    margin-right: ${px(unit)};
+    &:last-child {
+      margin-right: 0;
+    }
+  }
+`;
```

Comment:
You can remove all the custom styles if you use EUI flex components in the render function:
```suggestion
    <EuiFlexGroup alignItems="center">
      <EuiFlexItem grow={false}>
        <EuiTitle ... />
      </EuiFlexItem>
      {legends.map((legend) => (
        <EuiFlexItem grow={false} key={legend.value}>
          <Legend ... />
        </EuiFlexItem>
      ))}
    </EuiFlexGroup>
```

Reply:
Oh, that would be nice!

Reply:
nice one, thanks Oliver